src/physfs_platform_libretro.c

```diff
@@ -0,0 +1,421 @@
+/*
+ * libretro support routines for PhysicsFS.
+ *
+ * Please see the file LICENSE.txt in the source's root directory.
+ *
+ * This file written by Vladimir Serbinenko (@phcoder), and Rob Loach (@RobLoach).
+ */
+
+#define __PHYSICSFS_INTERNAL__
+#include "physfs_platforms.h"
+
+#ifdef PHYSFS_PLATFORM_LIBRETRO
+
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <errno.h>
+
+/* libretro dependencies */
+#include <libretro.h>
+#include <rthreads/rthreads.h>
```

Comment:
This currently depends on libretro's [`rthreads.c`](https://github.com/libretro/libretro-common/blob/b0c348ea5543c4d7fb0bc479258aa6988b20c0c9/rthreads/rthreads.c#L4), though there could be cases where you don't need it? Maybe introduce a `PHYSFS_PLATFORM_LIBRETRO_NO_THREADS` define to allow skipping its use?

Reply:
So we only need threading for mutexes, in case the app is multithreaded, but PhysicsFS doesn't spin any threads of its own, so if this isn't something that's generally needed, we can just not implement that part.

(but maybe rthreads is fine? I don't know the details on it.)

Reply:
Thanks for the clarification. Added `PHYSFS_PLATFORM_LIBRETRO_NO_THREADS` to allow disabling it.